Update broken test_data_set by the latest ORT

jcwchen opened the pull request:
**Description**
Update broken test_data_set by the latest ORT

P.S. Duplicate of https://github.com/onnx/models/pull/432. Create this new one to easily resolve ancient code conflict.

**Motivation**
Some test_data_set are broken: https://github.com/onnx/models/issues/431, https://github.com/onnx/models/issues/404, https://github.com/onnx/models/issues/403

garymm commented on workflow_scripts/check_model.py:
```diff
@@ -38,6 +45,8 @@ def run_backend_ort(model_path, test_data_set=None):
         onnxruntime.InferenceSession(model_path)
         ort_test_dir_utils.create_test_dir(model_path, './', test_utils.TEST_ORT_DIR)
         ort_test_dir_utils.run_test_dir(test_utils.TEST_ORT_DIR)
+        os.remove(tar_gz_path)
```
Why is this needed? Shouldn't `open` in mode `"w"` just overwrite it?

jcwchen replied:
Good point. I have removed it.

garymm commented on workflow_scripts/ort_test_dir_utils.py:
```diff
@@ -37,21 +38,24 @@ def _create_missing_input_data(model_inputs, name_input_map, symbolic_dim_values
-            elif dim_type == 'dim_param':
+            elif dim_type == "dim_param":
                 if dim.dim_param not in symbolic_dim_values_map:
+                    # The following two lines are different from onnxruntime
```
please expand the comment to explain why

garymm commented on workflow_scripts/test_models.py:
```diff
@@ -6,17 +6,26 @@
-                        help='Test the model by which (onnx/onnxruntime)?',
-                        choices=['onnx', 'onnxruntime', 'all'])
-    args = parser.parse_args()
+tar_ext_name = '.tar.gz'
```
use double or single quotes consistently in the file

garymm commented on workflow_scripts/test_models.py:
```diff
@@ -6,17 +6,26 @@
+def get_all_models():
+    model_list = []
+    parent_dir = ["text", "vision"]
+    for directory in parent_dir:
```
No need for a local variable here:
```suggestion
    for directory in ("text", "vision"):
```

garymm commented on workflow_scripts/test_models.py:
```diff
@@ -55,25 +81,40 @@ def main():
+                        print('[PASS] {} is checked by onnxruntime. '.format(model_name))
+                    except Exception as e:
+                        if not args.create:
+                            raise Exception(e)
```
No need to wrap the original exception if you're not adding any info or changing the type:
```suggestion
                            raise
```

garymm commented on workflow_scripts/test_utils.py:
```diff
@@ -49,11 +54,14 @@ def get_model_and_test_data(directory_path):
+        shutil.rmtree(directory_path)
+
+
 def remove_tar_dir():
```
I'd just replace these functions entirely with the one line of code that is inside the function.

Functions like this are not adding any value because they don't encapsulate anything useful.

garymm reviewed (approved):
Thanks!

garymm commented on workflow_scripts/check_model.py:
```diff
@@ -43,10 +44,13 @@ def run_backend_ort(model_path, test_data_set=None, tar_gz_path=None):
         onnxruntime.InferenceSession(model_path)
-        ort_test_dir_utils.create_test_dir(model_path, './', test_utils.TEST_ORT_DIR)
-        ort_test_dir_utils.run_test_dir(test_utils.TEST_ORT_DIR)
+        # Get model name without .onnx
```
Use https://docs.python.org/3/library/os.path.html#os.path.splitext and https://docs.python.org/3/library/os.path.html#os.path.basename

jcwchen commented:
Thank you @garymm for the reviews and I have updated the PR based on them.

Furthermore, there are some .tar.gz contain more than one ONNX model. (`squeezenet1.1-7.tar.gz, vgg16-7.tar.gz, vgg16-bn-7.tar.gz, vgg19-7.tar.gz, vgg19-bn-7.tar.gz, zfnet512-3.tar.gz, zfnet512-6.tar.gz, ResNet101-DUC-7.tar.gz`). I have removed redundant ones and updated them. Besides, now the new created .tar.gz will use model name as the directory name inside the .tar.gz.

After this PR is merged, all ONNX Model Zoo models with existing test_data_set can pass ORT with CPU provider, except bertsquad-8.onnx (the model is not runnable by ORT due to inference error, will file an issue for future tracking).

Since @garymm has reviewed/approved this PR, @wenbingl could you please sign-off this one to forward it? Thank you.

wenbingl commented on workflow_scripts/test_models.py:
```diff
@@ -6,81 +6,122 @@
     for model_path in model_list:
-        model_name = model_path.split('/')[-1]
-        print('==============Testing {}=============='.format(model_name))
+        model_name = model_path.split("/")[-1]
```
also enabling the 'black' formatting?

jcwchen replied:
Thanks for bringing this up. I was also thinking applying more Python lint/formatter in the CI and improving existing code. Will do this in another PR. At least for now, the added code in this PR pass flake8 except too long warnings.
